src/GameEngine/GameEngine.cpp

```diff
@@ -0,0 +1,161 @@
+//
+// Created by ker10 on 2019-10-23.
+//
+
+#include "GameEngine.h"
+#include <windows.h>
+#include "../Map/Map.h"
+#include "../MapLoader/MapLoader.h"
+#include "../Terminal/Terminal.h"
+#include "../exceptions.h"
+
+using std::to_string;
+
+namespace GameEngine {
+    GameEngine *GameEngine::game_engine_instance = nullptr;
+
+    GameEngine *GameEngine::instance() {
```

Comment:
```suggestion
    GameEngine *GameEngine::getInstance() {
```

Reply:
getInstance implies that it’s only a getter method, which this is not. I agree it kinda make sense, but if you see -getInstance()- you could easily think that just returns something call instance.

Reply:
Usually for a singleton type pattern like we are using here, it is assumed that it will return the instance of the object. It's not a big deal though, also my suggestion should actually be `get_instance` to conform to the naming convention